src/cr/cube/dimension.py

```diff
@@ -725,6 +734,214 @@ def _elements(self):
+                return self._subvar_aliases[self._raw_element_ids.index(_id)]
+        except ValueError:
+            return None
+
+        if _id >= 0 and _id < len(self._subvar_aliases):
+            return self._subvar_aliases[_id]
+
+        return None
+
+    @lazyproperty
+    def _raw_element_ids(self):
+        """Tuple of int or string element ids, as they appear in cube result
+
+        These are "raw" because they refer to the element ids before they've been
+        replaced with the alias for subvariables in the `._shimmed_dimension_dict`.
+        """
+        return tuple(
+            element["id"] for element in self._dimension_dict["type"]["elements"]
+        )
+
+    def _replaced_element_transforms(self, element_transforms):
+        """Replace the dictionary keys of element transforms with aliases
+
+        The element transforms identify which element they refer to by their key in
+        the element_transforms object. Before it is shimmed, this can identify them
+        in many different ways. The shim replaces these with the alias.
+        """
+        # --- The name of the element transform object is a string. This string is
+        # --- assumed to refer to the subvariable id, unless there is a "key", which
+        # --- can be "alias", in which case it refers to the alias, or if no
+        # --- subvariable is found with the id, in which case it's assumed to be the
+        # --- element_id
+        # --- TODO: This logic about "key" is not supported by the validation
+        # --- on the deck schema, so maybe we should remove it?
+        key = element_transforms.get("key")
+        if key == "alias":
+            # --- Already keyed by alias, no changes needed
+            return element_transforms
+
+        old_keys = tuple(element_transforms.keys())
+
+        if key == "subvar_id":
+            # --- translate from subvariable id
+            new_keys = tuple(
+                self._subvar_aliases[self._subvar_ids.index(_id)]
+                if _id in self._subvar_ids
+                else None
+                for _id in old_keys
+            )
+        else:
+            # --- Otherwise use usual translation logic
+            new_keys = tuple(self.translate_element_id(_id) for _id in old_keys)
+
+        return {
+            nkey: element_transforms[old_keys[i]]
+            for i, nkey in enumerate(new_keys)
+            if nkey is not None
+        }
+
+    def _replaced_order_element_ids(self, element_ids):
+        """Replace the list list of element ids with a list of aliases
+
+        The explicit order transform includes a list of ids that can be specified in
+        many different ways, this translate them to the subvariable aliases.
+        """
+        return [self.translate_element_id(_id) for _id in element_ids]
+
+    @lazyproperty
+    def _subvar_aliases(self):
+        """Tuple of string aliases for each element of a subvariable dimension
```

Comment:
Just a note on this style of summary string. It is `{sequence-type} of {item-type} {singular-item-name} for each {one-to-one-corresponding-thing}`.

So this would be `tuple of str alias for each element of a subvariable dimension.`

* `tuple` is a type and so is spelled as its type, even though it's the first word in the "sentence".
* `str` is a type, so we are as explicit as we can be, `str` here instead of `string`.
* `alias` is singular, rather than aliases.

You could think of it as: "tuple consisting of the str alias for each element of ...". We just leave out the consisting of and preposition "the" for brevity.

It's a fine point, but that's how it got the way you see it around, even though it takes a second sometimes to read it as intended.